### ISSUE TITLE
Missing links should fail silent in the frontend

### ISSUE THREAD

**Issue by rolandschuetz** [jonnitto/Carbon.Link#1]
Steps to reproduce:
 - You have a NodeType with a link property which links to a page B
 - When you create the current page (A) in a new dimension and page B doesn't exist in that dimension the node will be missing. In the backend, this is shown with "node missing".

Without carbon/link the link rendering would silently fail. With carbon/link the page will have no content in the frontend, because an exception is thrown:

```
Exception #1332491864 in line 202 of /data/releases/current/Data/Temporary/Production/Cache/Code/Flow_Object_Classes/Neos_Eel_FlowQuery_FlowQuery.php: No operation which satisfies the runtime constraints found for "closest".

124 Neos\Eel\FlowQuery\OperationResolver_Original::resolveOperation("closest", array|1|)
123 Neos\Eel\FlowQuery\FlowQuery_Original::evaluateOperations()
122 Neos\Eel\FlowQuery\FlowQuery_Original::__call("get", array|1|)
121 call_user_func_array(array|2|, array|1|)
120 Neos\Eel\Context::call("get", array|1|)
119 Neos\Eel\ProtectedContext::call("get", array|1|)
118 Neos\Eel\Context::callAndWrap("get", array|1|)
117 expression_d7836c81720bb7e9273ba5d9a38b7350(Neos\Eel\ProtectedContext)
116 Neos\Eel\CompilingEvaluator_Original::evaluate("q(nodeObject).closest(Configuration.setting('Carbon.Link.instanceOfDocument')).get(0)", Neos\Eel\ProtectedContext)
115 Neos\Eel\Utility_Original::evaluateEelExpression("${q(nodeObject).closest(Configuration.setting('Carbon.Link.instanceOfDocument')).get(0)}", Neos\Eel\CompilingEvaluator, array|36|)
114 Neos\Fusion\Core\Runtime_Original::evaluateEelExpression("${q(nodeObject).closest(Configuration.setting('Carbon.Link.instanceOfDocument')).get(0)}", Neos\Neos\Fusion\NodeUriImplementation)
113 Neos\Fusion\Core\Runtime_Original::evaluateEelExpressionOrSimpleValueWithProcessor("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...s.Fusion:Matcher>/renderer<Neos.Neos:NodeUri>/node", array|3|, Neos\Neos\Fusion\NodeUriImplementa
112 Neos\Fusion\Core\Runtime_Original::evaluteExpressionOrValueInternal("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...s.Fusion:Matcher>/renderer<Neos.Neos:NodeUri>/node", array|3|, array|5|, Neos\Neos\Fusion\NodeUriImplementation)
111 Neos\Fusion\Core\Runtime_Original::evaluateInternal("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...s.Fusion:Matcher>/renderer<Neos.Neos:NodeUri>/node", "NULL", Neos\Neos\Fusion\NodeUriImplementation)
110 Neos\Fusion\Core\Runtime_Original::evaluate("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...s.Fusion:Matcher>/renderer<Neos.Neos:NodeUri>/node", Neos\Neos\Fusion\NodeUriImplementation)
109 Neos\Fusion\FusionObjects\AbstractFusionObject::fusionValue("node")
108 Neos\Neos\Fusion\NodeUriImplementation_Original::getNode()
107 Neos\Neos\Fusion\NodeUriImplementation_Original::evaluate()
106 Neos\Fusion\Core\Runtime_Original::evaluateObjectOrRetrieveFromCache(Neos\Neos\Fusion\NodeUriImplementation, "root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher....Fusion:Case>/isNode<Neos.Fusion:Matcher>/renderer", array|9|, array|5|
105 Neos\Fusion\Core\Runtime_Original::evaluateInternal("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher....Fusion:Case>/isNode<Neos.Fusion:Matcher>/renderer", "NULL", Neos\Fusion\FusionObjects\MatcherImplementation)
104 Neos\Fusion\Core\Runtime_Original::evaluate("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher....Fusion:Case>/isNode<Neos.Fusion:Matcher>/renderer", Neos\Fusion\FusionObjects\MatcherImplementation)
103 Neos\Fusion\FusionObjects\RendererImplementation_Original::evaluate()
102 Neos\Fusion\FusionObjects\MatcherImplementation_Original::evaluate()
101 Neos\Fusion\Core\Runtime_Original::evaluateObjectOrRetrieveFromCache(Neos\Fusion\FusionObjects\MatcherImplementation, "root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...alue<Neos.Fusion:Case>/isNode<Neos.Fusion:Matcher>", array|4|,
100 Neos\Fusion\Core\Runtime_Original::evaluateInternal("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...alue<Neos.Fusion:Case>/isNode<Neos.Fusion:Matcher>", "Exception")
99 Neos\Fusion\Core\Runtime_Original::render("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...alue<Neos.Fusion:Case>/isNode<Neos.Fusion:Matcher>")
98 Neos\Fusion\FusionObjects\CaseImplementation_Original::renderMatcher("isNode")
97 Neos\Fusion\FusionObjects\CaseImplementation_Original::evaluate()
96 Neos\Fusion\Core\Runtime_Original::evaluateObjectOrRetrieveFromCache(Neos\Fusion\FusionObjects\CaseImplementation, "root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...arbon.Link:Attributes>/href<Carbon.Link:URI>/value", array|8|, arr
95 Neos\Fusion\Core\Runtime_Original::evaluateInternal("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...arbon.Link:Attributes>/href<Carbon.Link:URI>/value", "NULL", Neos\Fusion\FusionObjects\ValueImplementation)
94 Neos\Fusion\Core\Runtime_Original::evaluate("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...arbon.Link:Attributes>/href<Carbon.Link:URI>/value", Neos\Fusion\FusionObjects\ValueImplementation)
93 Neos\Fusion\FusionObjects\AbstractFusionObject::fusionValue("value")
92 Neos\Fusion\FusionObjects\ValueImplementation_Original::getValue()
91 Neos\Fusion\FusionObjects\ValueImplementation_Original::evaluate()
90 Neos\Fusion\Core\Runtime_Original::evaluateObjectOrRetrieveFromCache(Neos\Fusion\FusionObjects\ValueImplementation, "root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...ts.Teaser>/attributes<Carbon.Link:Attributes>/href", array|9|, ar
89 Neos\Fusion\Core\Runtime_Original::evaluateInternal("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...ts.Teaser>/attributes<Carbon.Link:Attributes>/href", "NULL", Neos\Fusion\FusionObjects\AttributesImplementation)
88 Neos\Fusion\Core\Runtime_Original::evaluate("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...ts.Teaser>/attributes<Carbon.Link:Attributes>/href", Neos\Fusion\FusionObjects\AttributesImplementation)
87 Neos\Fusion\FusionObjects\AbstractFusionObject::fusionValue("href")
86 Neos\Fusion\FusionObjects\AttributesImplementation_Original::evaluate()
85 Neos\Fusion\Core\Runtime_Original::evaluateObjectOrRetrieveFromCache(Neos\Fusion\FusionObjects\AttributesImplementation, "root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...r>/teaser<CodeQ.Site:Components.Teaser>/attributes", array|1
84 Neos\Fusion\Core\Runtime_Original::evaluateInternal("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...r>/teaser<CodeQ.Site:Components.Teaser>/attributes", "NULL", CodeQ\Site\FusionObjects\TemplateImplementation)
83 Neos\Fusion\Core\Runtime_Original::evaluate("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...r>/teaser<CodeQ.Site:Components.Teaser>/attributes", CodeQ\Site\FusionObjects\TemplateImplementation)
82 Neos\Fusion\FusionObjects\Helpers\FusionPathProxy_Original::objectAccess()
81 Neos\FluidAdaptor\Core\ViewHelper\TemplateVariableContainer::getByPath("attributes", array|0|)
80 Frontend_Node_action_show_d7a8187e398f096f43c5eb20c654b3eb2d96cb24::{closure}()
79 Frontend_Node_action_show_d7a8187e398f096f43c5eb20c654b3eb2d96cb24::render(Neos\FluidAdaptor\Core\Rendering\RenderingContext)
78 TYPO3Fluid\Fluid\View\AbstractTemplateView::render()
77 CodeQ\Site\FusionObjects\TemplateImplementation_Original::evaluate()
76 Neos\Fusion\Core\Runtime_Original::evaluateObjectOrRetrieveFromCache(CodeQ\Site\FusionObjects\TemplateImplementation, "root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...:Teaser.Section>/teaser2<CodeQ.Site:Teaser>/teaser", array|26|,
75 Neos\Fusion\Core\Runtime_Original::evaluateInternal("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...:Teaser.Section>/teaser2<CodeQ.Site:Teaser>/teaser", "NULL", CodeQ\Site\FusionObjects\TemplateImplementation)
74 Neos\Fusion\Core\Runtime_Original::evaluate("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...:Teaser.Section>/teaser2<CodeQ.Site:Teaser>/teaser", CodeQ\Site\FusionObjects\TemplateImplementation)
73 Neos\Fusion\FusionObjects\Helpers\FusionPathProxy_Original::objectAccess()
72 Neos\FluidAdaptor\Core\ViewHelper\TemplateVariableContainer::getByPath("teaser", array|0|)
71 Frontend_Node_action_show_35cf4d8514718a37a9d1d6e7df57ab1d18283a78::{closure}()
70 Frontend_Node_action_show_35cf4d8514718a37a9d1d6e7df57ab1d18283a78::render(Neos\FluidAdaptor\Core\Rendering\RenderingContext)
69 TYPO3Fluid\Fluid\View\AbstractTemplateView::render()
68 CodeQ\Site\FusionObjects\TemplateImplementation_Original::evaluate()
67 Neos\Fusion\Core\Runtime_Original::evaluateObjectOrRetrieveFromCache(CodeQ\Site\FusionObjects\TemplateImplementation, "root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...atcher>/element<CodeQ.Site:Teaser.Section>/teaser2", array|15|,
66 Neos\Fusion\Core\Runtime_Original::evaluateInternal("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...atcher>/element<CodeQ.Site:Teaser.Section>/teaser2", "NULL", CodeQ\Site\FusionObjects\TemplateImplementation)
65 Neos\Fusion\Core\Runtime_Original::evaluate("root<Neos.Fusion:Case>/default<Neos.Fusion:Matcher...atcher>/element<CodeQ.Site:Teaser.Section>/teaser2", CodeQ\Site\FusionObjects\TemplateImplementation)
64 Neos\Fusion\FusionObjects\Helpers\FusionPathProxy_Original::objectAccess()
63 Neos\FluidAdaptor\Core\ViewHelper\TemplateVariableContainer::getByPath("teaser2", array|0|)
62 Frontend_Node_action_show_9b8c6b2956a7ca8c25775cb00f956ffb47020e00::{closure}()
```

**Comment by jonnitto** [comment]
@rolandschuetz Can you paste me also your content dimension settings into the issue?

**Comment by rolandschuetz** [comment]
@jonnitto 
```
  ContentRepository:
    contentDimensions:
      language:
        label: Sprache
        icon: icon-language
        default: en
        defaultPreset: de
        presets:
          en:
            label: English
            values:
              - en
            uriSegment: en
          de:
            label: Deutsch
            values:
              - de
            uriSegment: de
          nl:
            label: Niederländisch
            values:
              - nl
            uriSegment: nl
          fr:
            label: Französisch
            values:
              - fr
            uriSegment: fr
          it:
            label: Italienisch
            values:
              - it
            uriSegment: it
          sv:
            label: Schwedisch
            values:
              - sv
            uriSegment: sv
          fi:
            label: Finnisch
            values:
              - fi
            uriSegment: fi
          es:
            label: Spanisch
            values:
              - es
            uriSegment: es
```

**Comment by jonnitto** [comment]
Solved with Version [0.4.0](https://github.com/jonnitto/Carbon.Link/releases/tag/0.4.0)
